master/internal/authz/obfuscate.go

```diff
@@ -90,6 +92,30 @@ func ObfuscateAgent(agent *agentv1.Agent) error {
 	return nil
 }
 
+// ObfuscateNTSCTask obfuscates sensitive information about a given Notebook, Tensorboard, Shell,
+// or Command Task.
+func ObfuscateNTSCTask(ntscTask *sproto.AllocationSummary) error {
```

Comment:
I think this is ok but just curious if we might want to obfuscate any of the int values like slots, priority, port number, etc. Or in resources the agent map. cc @corban-beaird

Reply:
For posterity, I'm making sure I include the note I gave Amanda on slack:
> **api/v1/tasks:**
> - Obfuscate all allocationIDs
> - In agentDevices: obfuscate the (ip) address being used as a key
> - In devices: hide the brand & uuid 
> - In proxyPorts: hide port, proxyTCP, and unauthenticated
> 
> _If it's being used as a key in a map, I'd recommend swapping that entry into a new one where the key is just an incrementing int_

I think basing what information we share so that it's consistent with the cluster overview page would be sufficient (assuming that we are using a user that has no role in an EE deployment